src/components/FeatureSetsPanel/FeatureSetsPanel.js

```diff
@@ -196,17 +206,13 @@ const FeatureSetsPanel = ({
             <FormSpy
               subscription={{ valid: true }}
               onChange={() => {
-                setValidation(prevState => ({
-                  ...prevState,
-                  areLabelsValid: formRef.current?.getFieldState?.('labels')?.valid ?? true
-                }))
+                setTempValidation(formRef.current?.getFieldState?.('labels')?.valid ?? true)
```

Comment:
Instead of creating a new temp state try to compare the new and the old `areLabelsValid` value and set the new validation object only if this value was changed